frontend/src/scenes/instance/SystemStatus/InternalMetricsTab.tsx

```diff
@@ -1,15 +1,115 @@
+                    </Collapse.Panel>
+                ) : null}
+                <Collapse.Panel header="PostgreSQL - currently running queries" key="1">
+                    <div className="mb float-right">
+                        <Checkbox
+                            checked={showIdle}
+                            onChange={(e) => {
+                                setShowIdle(e.target.checked)
+                            }}
+                        >
+                            Show idle queries
+                        </Checkbox>
+                        <Button style={{ marginLeft: 8 }} onClick={reloadQueries}>
+                            <ReloadOutlined /> Reload Queries
+                        </Button>
+                    </div>
+                    <QueryTable queries={postgresQueries} loading={queriesLoading} />
+                </Collapse.Panel>
+                {queries?.clickhouse_running != undefined ? (
+                    <Collapse.Panel header="Clickhouse - currently running queries" key="2">
+                        <div className="mb float-right">
+                            <Button style={{ marginLeft: 8 }} onClick={reloadQueries}>
+                                <ReloadOutlined /> Reload Queries
+                            </Button>
+                        </div>
+                        <QueryTable queries={queries?.clickhouse_running} loading={queriesLoading} />
+                    </Collapse.Panel>
+                ) : null}
+                {queries?.clickhouse_slow_log != undefined ? (
+                    <Collapse.Panel header="Clickhouse - slow query log (past 6 hours)" key="3">
+                        <div className="mb float-right">
+                            <Button style={{ marginLeft: 8 }} onClick={reloadQueries}>
+                                <ReloadOutlined /> Reload Queries
+                            </Button>
+                        </div>
+                        <QueryTable queries={queries?.clickhouse_slow_log} loading={queriesLoading} />
+                    </Collapse.Panel>
+                ) : null}
+            </Collapse>
+        </Card>
+    )
+}
+
+function QueryTable(props: {
+    queries?: QuerySummary[]
+    loading: boolean
+    columnExtra?: Record<string, any>
+}): JSX.Element {
+    const columns: ColumnsType<QuerySummary> = [
+        {
+            title: 'duration',
+            dataIndex: 'duration',
+            key: 'duration',
+            sorter: (a, b) => +a.duration - +b.duration,
+        },
+        {
+            title: 'query',
+            dataIndex: 'query',
+            key: 'query',
+        },
+    ]
+
+    if (props.queries && props.queries.length > 0) {
+        Object.keys(props.queries[0]).forEach((column) => {
+            if (column !== 'duration' && column !== 'query') {
+                columns.push({ title: column, dataIndex: column, key: column })
+            }
+        })
+    }
```

Comment:
Maybe consider sth like this (not valid code as I don't know Typescript all that well, but the idea):
```
const columns: ColumnsType<QuerySummary> = []
first_cols = ['duration', 'query']
first_cols.forEach((column) => {
  columns.push({ title: column, dataIndex: column, key: column })
}
columns[0].sorter = (a,b) => +a.duration - +b.duration

if (props.queries && props.queries.length > 0) {
        Object.keys(props.queries[0]).forEach((column) => {
            if (column not in first_cols) {
                columns.push({ title: column, dataIndex: column, key: column })
            }
        })
    }
```

Reply:
Valid idea. I'd like to keep query + duration as explicit though since while your suggestion is shorter, we might want to add a custom render method. Also this is more inline with rest of our tables (though we usually have less dynamic columns)